Release v6.0.0 and README update (#29)
camunda/camunda-modeler-plugin-helpers: README.md

registerBpmnJSModdleExtension(moddleDescriptor);
```

> [![Compatible with Camunda Modeler version 5.29+](https://img.shields.io/badge/Camunda%20Modeler-5.29+-blue.svg)](https://github.com/camunda/camunda-modeler)

Reviewer: This looks out of place:

![image](https://github.com/user-attachments/assets/fc4bc542-72a5-4cd2-9fea-2a6b5014198d)

Maybe you want to wrap this in a NOTE block instead?

Alternatively we could change the error messages thrown to be clear on the environments that initially added support for something.

===> You test your plug-in in a Modeler, it :boom:'s with `requires Camunda Modeler >= v5.33.0`.

Author: Added in https://github.com/camunda/camunda-modeler-plugin-helpers/commit/df5f787c45f63833375b24feab2603974f83acc1

https://github.com/camunda/camunda-modeler-plugin-helpers/blob/df5f787c45f63833375b24feab2603974f83acc1/README.md?plain=1#L48

## UI Components
These components are available for use:

- **Fill**
- **Modal**
- **Overlay** 
- **Section**
- **ToggleSwitch**
- **TextInput**
- **CachedComponent**

Components exported for use in Camunda Modeler client plugins.

```javascript

> import { Fill } from 'camunda-modeler-plugin-helpers/components';

Reviewer: This is not a valid JS snippet, let's fix it.

```suggestion
import { Fill } from 'camunda-modeler-plugin-helpers/components';

function MyReactExtension() {
  return <Fill ... />
};
```
```

Author: Fixed.

- **WithCachedState**  
  *A higher-order component that lazily initiates the given wrapped component.*

## Helper Functions

- **createTab**  
  *A helper function to create Tab components to be used with the TabProvider via the `Comp#createCachedState` method.*

## Properties Panel

The following packages are exposed and can be used with `CamundaModelerWebpackPlugin` to configure bundling for the client:

In your webpack configuration, set aliases:

```javascript
// webpack.config.js
  plugins: [
    new CamundaModelerWebpackPlugin({
      type: '@bpmn-io/properties-panel'
    }),
    new CamundaModelerWebpackPlugin({
      type: 'bpmn-js-properties-panel'
    }),
    new CamundaModelerWebpackPlugin({
      type: 'dmn-js-properties-panel'
    })
  ]
```

Then, in the source code of your extension, you can still import:

```js
import { useService } from 'bpmn-js-properties-panel';
import { useState } from '@bpmn-io/properties-panel/preact/hooks';
```

## Packages

[![Compatible with Camunda Modeler version 5.33+](https://img.shields.io/badge/Camunda%20Modeler-5.33+-blue.svg)](https://github.com/camunda/camunda-modeler)

> The following packages are exposed and can be used with `CamundaModelerWebpackPlugin` to configure bundling for the client:

Reviewer: Proposal: Don't showcase webpack plug-in configuration here, but link it promintently on top of the library README.

Reviewer: :arrow_right: https://github.com/camunda/camunda-modeler-plugin-helpers/pull/29#issuecomment-2711142249

Author: Haven't made any changes to the webpack plug-in linking as this is already implemented in https://github.com/camunda/camunda-modeler-plugin-helpers/pull/30. 
Waiting for it to merged, so we can adjust the current packages link i.e., `@carbon/react`

Additionally, I've removed explicit webpack plugin config. https://github.com/camunda/camunda-modeler-plugin-helpers/commit/df5f787c45f63833375b24feab2603974f83acc1#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L119-L133

Author: Since #30 is merged, I've updated the current packages link. 

* UI components 
* `vendor/properties-panel`
* `vendor/@carbon/react`

```javascript

> // webpack.config.js

Reviewer: Again, this is not a valid snippet, let's fix it.

Author: fixed

- **Fill**
- **Modal**
- **Overlay** 
- **Section**
- **ToggleSwitch**
- **TextInput**
- **CachedComponent**

Components exported for use in Camunda Modeler client plugins.

```javascript
import { Fill } from 'camunda-modeler-plugin-helpers/components';
    <Fill slot="status-bar__app" group="1_first">
      <button
        className="btn"
        type="button"
        onClick={ () => setModalOpen(true) }
      >
        Test 
      </button>
```

## Higher-Order Function Helpers
These higher-order components (HOCs) enhance functionality:

- **WithCache**  
  *A higher-order component that passes cache to a wrapped component. Forwards refs, too.*

- **WithCachedState**  
  *A higher-order component that lazily initiates the given wrapped component.*

## Helper Functions

- **createTab**  
  *A helper function to create Tab components to be used with the TabProvider via the `Comp#createCachedState` method.*

## Properties Panel

Use libraries exposed in Camunda Modeler via /vendor subdirectories:
  * /vendor/@bpmn-io/properties-panel/...
  * /vendor/bpmn-js-properties-panel
  * /vendor/react

In your webpack configuration, set aliases:

```js
// webpack.config.js

> {

Reviewer: Instead of doing this, let's encourage use of the webpack plug-in?

Author: Done.

Author: will add properties panel related config in `camunda-modeler-webpack-plugin` once this is merged.

registerBpmnJSModdleExtension(moddleDescriptor);
```
> [!NOTE]

> > This feature requires **Camunda Modeler 5.29+**.

Reviewer 2: Does it apply to the feature above or below?

Author: Below. adjusted.

## UI Components
These components are available for use:

> - **Fill**

Reviewer 2: We should link those to the corresponding file in https://github.com/camunda/camunda-modeler-plugin-helpers/tree/main/components, cause there are some examples.

- **Modal**
- **Overlay** 
- **Section**
- **ToggleSwitch**
- **TextInput**
- **CachedComponent**

Components exported for use in Camunda Modeler client plugins.

```javascript

> import Fill from 'camunda-modeler-plugin-helpers/components/Fill.js';

Reviewer: This relies on #31, which we have not merged (yet). I'd consider to not document this (at this stage).

Reviewer: Alternatively consider this PR blocked until we verify that #31 works and is merged.

Reviewer 3: I'll convert this PR to draft since #31 is a draft.

  function MyReactExtension() {
    return <Fill ... />
};
```

## Higher-Order Function Helpers
These higher-order components (HOCs) enhance functionality:

- **WithCache**  
  *A higher-order component that passes cache to a wrapped component. Forwards refs, too.*

- **WithCachedState**  

>   *A higher-order component that lazily initiates the given wrapped component.*

Reviewer 2: ```suggestion
- **[WithCache](https://github.com/camunda/camunda-modeler-plugin-helpers/blob/main/components/WithCache.js)** - pass cache and forward refs to a wrapped component.

- **[WithCachedState](https://github.com/camunda/camunda-modeler-plugin-helpers/blob/main/components/WithCachedState.js)** - lazy-load a wrapped component.
```

## Helper Functions

- **createTab**  

>   *A helper function to create Tab components to be used with the TabProvider via the `Comp#createCachedState` method.*

Reviewer 2: ```suggestion
- **[createTab](https://github.com/camunda/camunda-modeler-plugin-helpers/blob/main/components/create-tab.js)** - helper function to create Tab components to be used with the TabProvider via the `Comp#createCachedState` method.
```

import React, { useEffect } from 'camunda-modeler-plugin-helpers/vendor/react.js';
```

### Carbon

> [!NOTE]

> > This feature requires **Camunda Modeler 5.38+**.
> > Should be used in place of `@carbon/react` imports.

Reviewer 2: ```suggestion
> This feature requires **Camunda Modeler 5.38+**.

> Should be used in place of `@carbon/react` imports.
```

It renders in one line otherwise.

Author: I think it's fine as far as the message is clear.
